packages/ragbits-core/src/ragbits/core/vector_stores/chroma.py

```diff
@@ -165,17 +168,77 @@ async def list(
         embeddings = results.get("embeddings") or []
         documents = results.get("documents") or []
         metadatas = (
-            [json.loads(metadata["__metadata"]) for metadata in results.get("metadatas", [])]  # type: ignore
-            if self._metadata_store is None
-            else await self._metadata_store.get(ids)
+            results.get("metadatas") or [] if self._metadata_store is None else await self._metadata_store.get(ids)
         )
 
         return [
             VectorStoreEntry(
                 id=id,
                 key=document,
                 vector=list(embedding),
-                metadata=metadata,  # type: ignore
+                metadata=self._unflatten_dict(metadata) if metadata else {},  # type: ignore
             )
             for id, metadata, embedding, document in zip(ids, metadatas, embeddings, documents, strict=True)
         ]
+
+    @staticmethod
+    def _flatten_dict(d: dict[str, Any], parent_key: str = "", sep: str = ".") -> dict[str, str | int | float | bool]:
+        """
+        Recursively flatten a nested dictionary, converting non-primitive types to strings.
+
+        Args:
+            d: The dictionary to flatten
+            parent_key: The parent key for nested keys
+            sep: The separator between nested keys
+
+        Returns:
+            A flattened dictionary with primitive types and stringified complex types
+        """
+        items: dict[str, str | int | float | bool] = {}
+        for k, v in d.items():
+            new_key = f"{parent_key}{sep}{k}" if parent_key else k
+
+            if isinstance(v, dict):
+                items = {**items, **ChromaVectorStore._flatten_dict(v, new_key, sep=sep)}
+            else:
+                items[new_key] = v if isinstance(v, str | int | float | bool) else str(v)
```

Comment:
We shouldn't store any complex types in the metadata, because we won't be able to reproduce them during unflattening, which will cause a validation error when parsing to the `Element` class. 

Support for `None` types should also be added. For example, if we convert the value to ``None`` and the field is typed as an optional string, we will incorrectly recreate the metadata.

Reply:
Strings with `[]` in them will not be treated correctly later on - they'll be considered lists.